enos/enos-scenario-upgrade.hcl

```diff
@@ -698,6 +698,31 @@ scenario "upgrade" {
     }
   }
 
+  step "verify_billing_start_date" {
+    description = global.description.verify_billing_start_date
+    skip_step   = matrix.edition == "ce"
```

Comment:
If I understand this correctly, this will only apply to certain versions of Vault. We'll probably want to guard it here based on a semver constraint,  e.g: https://github.com/hashicorp/vault/blob/main/enos/enos-scenario-autopilot.hcl#L97

Reply:
This change will be merged to main, 1.17 and 1.16 . So upgrades from any version to the latest version in the branch will have an auto-rolled billing start. I don't think we need to protect against a version as upgrades can happen from any version to the version of the branch if i understand correctly.

Reply:
I understand now, is this product_version the upgrade version?